src/group_theory/torsion.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2022 Julian Berman. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Berman
+-/
+
+import group_theory.exponent
+import group_theory.order_of_element
+import group_theory.quotient_group
+
+/-!
+# Torsion groups
+
+This file defines torsion groups, i.e. groups where all elements have finite order.
+
+## Main definitions
+
+* `monoid.is_torsion` is a predicate asserting a monoid `G` is a torsion monoid, i.e. that all
+  elements are of finite order. Torsion groups are also known as periodic groups.
+* `add_monoid.is_torsion` the additive version of `monoid.is_torsion`.
+
+## Main results
+
```

Comment:
A TODO for later could be to define the torsion subgroup of a group (see e.g. [here](https://encyclopediaofmath.org/wiki/Torsion_group)), prove that it is a normal subgroup and define the torsion-free quotient group.

Reply:
Thanks! Makes sense, can do -- so far I found/was reading "A Course in the Theory of Groups" (Robinson) which also has a few sections mentioning these (and looked like also has some other kinds of groups defined in terms of them).

Reply:
(I added your suggestions as notes -- hopefully that was the right thing to do.)

Also I see there are some things mentioning torsion-free modules defined in `linear_algebra/free_module/pid.lean` -- are these related / if so presumably I should try to generalize what's there from a ring to the additive group?

Reply:
An `add_comm_group` is the same as a Z-module. In this case the concept of a torsion (sub)group and of a torsion (sub)module agree. So the two concepts are generalisations in two different directions (either generalising by removing the `comm` condition or by generalising the ring.

Reply:
Aha, I see, thanks!